File: strings/796. Rotate String/rotateString.py
class Solution(object):
    def rotateString(self, s, goal):
        """
        :type s: str
        :type goal: str
        :rtype: bool
        """
        for i in range(len(s)):
            first_char = s[0]
            s = s [1:]
            s += first_char
            if s == goal:
                return True
        return False

File: strings/796. Rotate String/rotateString.js
/**
 * @param {string} s
 * @param {string} goal
 * @return {boolean}
 */
var rotateString = function(s, goal) {
    for (let i = 0; i < s.length; i++) {
        let firstChar = s[0];
        s = s.slice(1);
        s += firstChar;
        if (s === goal) return true;
    }
    return false;
};